client/AdminClient.c

```diff
@@ -119,11 +119,14 @@ static Iface_DEFUN receiveMessage(struct Message* msg, struct Iface* addrIface)
 
     struct Sockaddr_storage source;
     Message_pop(msg, &source, ctx->targetAddr->addrLen, NULL);
-    if (Bits_memcmp(&source, ctx->targetAddr, ctx->targetAddr->addrLen)) {
+    if (Bits_memcmp(&source+1, ctx->targetAddr+1, ctx->targetAddr->addrLen-1)) {
```

Comment:
You're skipping the wrong byte and you're doing it for all platforms, including those which do not have it. This byte is also in OSX and FreeBSD so you need to find where it is being properly cleared for those platforms (it is ok to clear the byte, it does no harm for it to be zero)